extension/src/main/kotlin/impl/RemoteHistoryService.kt

```diff
@@ -0,0 +1,43 @@
+/*-
+ * #%L
+ * camunda-rest-client-spring-boot
+ * %%
+ * Copyright (C) 2021 Camunda Services GmbH
+ * %%
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ *  under one or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information regarding copyright
+ *  ownership. Camunda licenses this file to you under the Apache License,
+ *  Version 2.0; you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * #L%
+ */
+package org.camunda.bpm.extension.rest.impl
+
+import org.camunda.bpm.extension.rest.adapter.AbstractHistoryServiceAdapter
+import org.camunda.bpm.extension.rest.client.api.HistoricProcessInstanceApiClient
+import org.camunda.bpm.extension.rest.impl.query.DelegatingHistoricProcessInstanceQuery
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Component
+
+/**
+ * Remote implementation of Camunda Core HistoryService API, delegating
+ * all request over HTTP to a remote Camunda Engine.
+ */
+@Component
+@Qualifier("remote")
+class RemoteHistoryService(
```

Comment:
Why do you need it for this PR? Is it not a separate issue (just created) #128

Reply:
Yes you are probably right. I wanted to include all parameters and the process instances to suspend/activate can be queried by a historic process instance query.